contracts/src/interfaces/IStakeTable.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: Unlicensed
+
+pragma solidity ^0.8.0;
+
+import { BN254 } from "bn254/BN254.sol";
+import { EdOnBN254 } from "../libraries/EdOnBn254.sol";
+
+/// @title Interface for stake table that keep track of validators' external key and staked amount.
+/// @dev Stake delegation happens in a separate DelegationPool contract, and specific to
+/// instantiation thus not part of this interface
+/// @dev Stake table contract should store a reference to the `LightClient.sol` to query
+/// "epoch-related" info
+interface IStakeTable {
+    /// @notice Supported stake type, either using native token or re-staked using ETH
+    enum StakeType {
+        Native,
+        Restake
+    }
+
+    /// @dev (sadly, Solidity doesn't support type alias on non-primitive types)
+    // We avoid declaring another struct even if the type info helps with readability,
+    // extra layer of struct introduces overhead and more gas cost.
+    //
+    /// @dev BLS verification key (used for Consensus voting in HotShot) is `BN254.G1Point`.
+    // `type BlsVerKey is BN254.G1Point;`
+    //
+    /// @dev Verification key of a Schnorr signature is just a `EdOnBN254.EdOnBN254Point`
+    // `type SchnorrVerKey is EdOnBN254.EdOnBN254Point;`
+
+    /// @notice Represents a HotShot validator node
+    /// In the dual-staking model, a HotShot validator could have multiple `Node` entries.
+    /// @param account The Ethereum account of the validator.
+    /// @param stakeType The type of token staked.
+    /// @param balance The amount of token staked.
+    /// @param registerEpoch The starting epoch for the validator.
+    /// @param exitEpoch The ending epoch for the validator.
+    /// @param schnorrVK The Schnorr verification key associated.
+    struct Node {
+        address account;
+        StakeType stakeType;
+        uint64 balance;
+        uint64 registerEpoch;
+        uint64 exitEpoch;
+        EdOnBN254.EdOnBN254Point schnorrVK;
+    }
+
+    // === Table State & Stats ===
+
+    /// @notice Get the total stakes of the registered keys in the latest stake table (Head).
+    /// @return The total stake for native token and restaked token respectively.
+    function totalStake() external view returns (uint256, uint256);
+    /// @notice Get the total number of stakers, uniquely identified by their `blsVK`
+    function totalKeys() external view returns (uint32);
+    /// @notice Get the total stake of the registered keys in the voting stake table
+    /// (LastEpochStart)
+    function totalVotingStake() external view returns (uint256);
+
+    /// @notice Look up the balance of `blsVK`
+    function lookup(BN254.G1Point calldata blsVK) external view returns (uint64);
+    /// @notice Look up the full `Node` state associated with `blsVK`
+    function fullLookup(BN254.G1Point calldata blsVK) external view returns (Node memory);
```

Comment:
Nit: I would prefer more descriptive names like `lookupStake` and `lookupNode`

Reply:
I like it, done in [612a578](https://github.com/EspressoSystems/espresso-sequencer/pull/748/commits/612a5787aa1ab5d81c73765aa22de498c06a3f23)